sonar-groovy-plugin/src/main/java/org/sonar/plugins/groovy/codenarc/CodeNarcSensor.java

```diff
@@ -100,89 +101,87 @@ public void analyse(Project project, SensorContext context) {
+      newIssue.at(location).save();
+    }
   }
 
-  private void runCodeNarc() {
+  private void runCodeNarc(SensorContext context) {
     LOG.info("Executing CodeNarc");
 
     File workdir = new File(fileSystem.workDir(), "/codenarc/");
     prepareWorkDir(workdir);
     File codeNarcConfiguration = new File(workdir, "profile.xml");
-    exportCodeNarcConfiguration(codeNarcConfiguration);
+    exportCodeNarcConfiguration(context, codeNarcConfiguration);
 
     CodeNarcRunner runner = new CodeNarcRunner();
     runner.setRuleSetFiles("file:" + codeNarcConfiguration.getAbsolutePath());
 
     CodeNarcSourceAnalyzer analyzer = new CodeNarcSourceAnalyzer(groovyFileSystem.sourceFiles());
     runner.setSourceAnalyzer(analyzer);
     runner.execute();
-    reportViolations(analyzer.getViolationsByFile());
+    reportViolations(context, analyzer.getViolationsByFile());
   }
 
-  private void reportViolations(Map<File, List<Violation>> violationsByFile) {
+  private void reportViolations(SensorContext context, Map<File, List<Violation>> violationsByFile) {
     for (Entry<File, List<Violation>> violationsOnFile : violationsByFile.entrySet()) {
-      Issuable issuable = issuableFor(violationsOnFile.getKey().getAbsolutePath());
-      if (issuable == null) {
+      InputFile groovyFile = inputFileFor(violationsOnFile.getKey().getAbsolutePath());
+      if (groovyFile == null) {
         continue;
       }
       for (Violation violation : violationsOnFile.getValue()) {
-        String ruleName = violation.getRule().getName();
-        RuleQuery ruleQuery = RuleQuery.create().withRepositoryKey(CodeNarcRulesDefinition.REPOSITORY_KEY).withConfigKey(ruleName);
-        Rule rule = ruleFinder.find(ruleQuery);
-        if (rule != null) {
-          insertIssue(rule.ruleKey(), violation.getLineNumber(), violation.getMessage(), issuable);
+        String ruleKey = violation.getRule().getName();
+        ActiveRule activeRule = context.activeRules().findByInternalKey(CodeNarcRulesDefinition.REPOSITORY_KEY, ruleKey);
+        if (activeRule != null) {
+          insertIssue(context, activeRule.ruleKey(), violation.getLineNumber(), violation.getMessage(), groovyFile);
         } else {
-          LOG.warn("No such rule in Sonar, so violation from CodeNarc will be ignored: ", ruleName);
+          LOG.warn("No such rule in Sonar, so violation from CodeNarc will be ignored: ", ruleKey);
         }
       }
     }
   }
 
   @CheckForNull
-  private Issuable issuableFor(String path) {
-    InputFile sonarFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(path));
-    if (sonarFile != null) {
-      return perspectives.as(Issuable.class, sonarFile);
-    }
-    return null;
+  private InputFile inputFileFor(String path) {
+    return fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(path));
   }
 
-  private void exportCodeNarcConfiguration(File file) {
+  private void exportCodeNarcConfiguration(SensorContext context, File file) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "context". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1172)